reconcile/utils/terrascript_aws_client.py

```diff
@@ -1926,6 +1930,27 @@ def populate_tf_resource_s3(self, spec):
         if cors_rules:
             # common_values['cors_rules'] is a list of cors_rules
             values["cors_rule"] = cors_rules
+        # S3 Bucket Logging
+        s3_bucket_logging = common_values.get("s3_bucket_logging")
+        if s3_bucket_logging:
+            target_bucket_name = s3_bucket_logging.get("target_bucket_name")
+            logging_identifier = f"{identifier}-logging"
+
+            # Lookup target S3 bucket by name
+            self.add_resource(
+                account,
+                data.aws_s3_bucket(target_bucket_name, bucket=target_bucket_name),
+            )
```

Comment:
why do we need to add this resource here if it has been defined elsewhere? Is it to check that it exists?

Reply:
It is a recommended practices that I picked up over time which is to use a data source to lookup and use a resource which is not part of the terraform config. In this case the s3_bucket_logging resource will fail if provided with a bucket that does not exist so this does not provide a lot of value at the increase cost of additional AWS API calls. This was used more in code which I since removed so I'll remove the data source as well